backend/src/main/java/com/bakdata/conquery/models/query/resultinfo/printers/ResultPrinters.java

```diff
@@ -0,0 +1,189 @@
+package com.bakdata.conquery.models.query.resultinfo.printers;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Objects;
+import java.util.StringJoiner;
+
+import com.bakdata.conquery.internationalization.Results;
+import com.bakdata.conquery.models.common.CDate;
+import com.bakdata.conquery.models.common.daterange.CDateRange;
+import com.bakdata.conquery.models.config.LocaleConfig;
+import com.bakdata.conquery.models.index.InternToExternMapper;
+import com.bakdata.conquery.models.query.PrintSettings;
+import com.bakdata.conquery.models.types.ResultType;
+import com.google.common.base.Preconditions;
+import lombok.ToString;
+import lombok.experimental.UtilityClass;
+import lombok.extern.slf4j.Slf4j;
+
+@UtilityClass
+@Slf4j
+public class ResultPrinters {
+
+	public Printer defaultPrinter(ResultType type) {
+		if (type instanceof ResultType.ListT<?> listT) {
+			return new ListPrinter(defaultPrinter(listT.getElementType()));
+		}
+
+		return switch (((ResultType.Primitive) type)) {
+			case BOOLEAN -> new BooleanPrinter();
+			case INTEGER -> new IntegerPrinter();
+			case NUMERIC -> new NumericPrinter();
+			case DATE -> new DatePrinter();
+			case DATE_RANGE -> new DateRangePrinter();
+			case STRING -> new StringPrinter();
+			case MONEY -> new MoneyPrinter();
+		};
+	}
+
+	public BigDecimal readMoney(PrintSettings cfg, Number value) {
+		return new BigDecimal(value.longValue()).movePointLeft(cfg.getCurrency().getDefaultFractionDigits());
+	}
+
+	@FunctionalInterface
+	public interface Printer {
+		String print(Object f, PrintSettings cfg);
+	}
```

Comment:
Die PrintSettings sollten nicht für jeden print Aufruf mitgegeben werden, sondern bei der Printer Instanziierung schon feststehen

Reply:
ist in der mache